frameworks/hdfs/tests/test_sanity.py

```diff
@@ -148,10 +147,23 @@ def test_kill_data_node():
 @pytest.mark.sanity
 @pytest.mark.recovery
 def test_kill_scheduler():
+    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
+    scheduler_name = sdk_utils.get_scheduler_name(foldered_name)
+    scheduler_id = sdk_tasks.get_task_ids("marathon", scheduler_name)
+    journal_ids = sdk_tasks.get_task_ids(foldered_name, "journal")
+    name_ids = sdk_tasks.get_task_ids(foldered_name, "name")
+    data_ids = sdk_tasks.get_task_ids(foldered_name, "data")
+
     sdk_cmd.kill_task_with_pattern(
-        "hdfs.scheduler.Main", shakedown.get_service_ips("marathon").pop()
+        "./hdfs-scheduler/bin/hdfs",
+        sdk_marathon.get_scheduler_host(foldered_name),
     )
-    config.check_healthy(service_name=sdk_utils.get_foldered_name(config.SERVICE_NAME))
+
+    sdk_tasks.check_tasks_updated("marathon", scheduler_name, scheduler_id)
+    sdk_tasks.check_tasks_not_updated(foldered_name, "journal", journal_ids)
+    sdk_tasks.check_tasks_not_updated(foldered_name, "name", name_ids)
+    sdk_tasks.check_tasks_not_updated(foldered_name, "data", data_ids)
+    config.check_healthy(service_name=foldered_name)
```

Comment:
Would it not be better to `check_healthy` **before** we check that none of the tasks have completed. This will give an unexpected deployment or recovery time to completed (and change the tasks) before we check that they have not changed.

Reply:
This was in Nick's PR so I copied it over but I'm not sure I follow you. `check_healthy` is just a bunch of gets against the deploy/recovery plan for a status `COMPLETE` and a check that the task is in state `TASK_RUNNING`. I don't think it changes any of the tasks?

To me it makes sense to check that the service is healthy after restarting the scheduler to verify that everything is still in working order?